### ISSUE TITLE
[crash][1.12.2, 1.15.2 pre18] shaders crash when joining world

### ISSUE THREAD

**Issue by MilkMGN** [sp614x/optifine#3981]
## Description of Issue
when i have a shader loaded and try to join a 1.12.2 modpack that i made for myself/ my vanilla 1.15.2 server, the game crashes. i can have the render distance on 2 and lowest graphics settings, and my game will still crash.

## Steps to Reproduce
1. dunno, this started randomly.
2. maybe have java 64 bit

## OptiFine Version
1.12.2 latest + 1.15.2 pre18 (probably every other version with shaders support

## Installation Method
both ways (installer and drag and drop)
## Fabric/Forge Version
when ran on 1.12.2  the forge version was 1.12.2-14.23.5.2847

## Other Installed Mods
https://pastebin.com/LPJFMQH0 (for the 1.12.2 crash)
**If OptiFine is installed as standalone, remove this section.**

## Log Files/Crash Reports
[debug.log](https://github.com/sp614x/optifine/files/4666759/debug.log)
[debug-1.log.gz](https://github.com/sp614x/optifine/files/4666760/debug-1.log.gz)
[latest.log](https://github.com/sp614x/optifine/files/4666761/latest.log)
[2020-05-22-7.log.gz](https://github.com/sp614x/optifine/files/4666762/2020-05-22-7.log.gz)
## F3 Debug Screenshot
N/A, cant get into game

## Prior Testing
If you've installed standalone:
- Does this happen in Vanilla without OptiFine?
no
If you've installed using Forge/Fabric:
- Does this happen without other mods installed? yes
   - Only OptiFine and Forge yes
- Does this happen without any mods at all? yes
- Does this happen in OptiFine standalone? yes
- Does this happen in Vanilla without OptiFine? no
- Have you tried using a binary search to find which mods cause this issue? yes
   - To find conflicting mods, split your mods into 2 groups (not including OptiFine/OptiFabric, obviously). Remove one group, and test in-game. Keep the group that has the problem, and repeat until no more mods can be removed without the issue disappearing.
   - Does the issue still happen with only the remaining mods installed?
      - Forge/Fabric and remaining mods, without OptiFine

## Additional Information
Here goes any final remarks and everything else that might not fit in the previous sections.

might be a bit weird to read, sorry for that,

**Comment by realjackasterisk** [comment]
Please use the binary search to find conflicting mods. It's listed in the issue template, which you have not finished editing.

**Comment by MilkMGN** [comment]
binary search should'nt matter as it is also happening on straight up vanilla.

**Comment by realjackasterisk** [comment]
ok, this is a very confusing issue to read.

Have you tried other shader packs? Can you post system specs? Are your drivers up to date? Does this happen in singleplayer/multiplayer/both?

**Comment by MilkMGN** [comment]
yes, ive tried other packs, system specs are i7 930, gtx 750 ti oc, 12 gigs ram (9 dedicated to minecraft), i updated the driver yesterday, and yes it happens in both

**Comment by naeqirelle2020** [comment]
@MilkMGN try dedicating 2GB to minecraft and seeing if it loads up, you look like you're dedicating too much ram to minecraft and that could be the source of your issues

**Comment by MilkMGN** [comment]
yeah it worked with 2gb, weird that it doesn't work with more, as PC's are becoming better and better, people are gonna want to allocate more to their Minecraft instance.
